Update release notes for 2022.13 release (#2703)
uxlfoundation/oneDPL: documentation/release_notes.rst

Reviewer: This limitation is now addressed, should be removed.

Author: Done.

- The sorting algorithms with device policies now can use radix sort with
  the ``less`` and ``greater`` comparators from the ``std::ranges`` namespace [#fnote1]_.
- Implemented support for bounded output for range-based set algorithms with CPU policies,
  with optional support for C++26 semantics of return values.
- Improved performance of ``sort``, ``stable_sort``, ``sort_by_key``, and ``stable_sort_by_key`` when using
  Radix sort [#fnote1]_ and device policies for key and/or value types larger than four bytes.
- Improved performance of ``inclusive_scan``, ``exclusive_scan``, ``transform_inclusive_scan``, and
  ``transform_exclusive_scan`` with device policies for non-trivially-copyable value types on GPU devices.
- Improved performance of the ``histogram_even`` and ``histogram_range`` algorithms with device policies for a small
  number of bins on GPU devices.

Known Issues and Limitations
----------------------------
Existing Issues
^^^^^^^^^^^^^^^
See oneDPL Guide for other `restrictions and known limitations`_.

- ``set_union``, ``set_intersection``, ``set_difference``, ``set_symmetric_difference`` range algorithms require the

>   output range to have sufficient size to hold all resulting elements.

Reviewer: ```suggestion
- When used with device policies. ``set_union``, ``set_intersection``, ``set_difference``, ``set_symmetric_difference``
  range algorithms require the output range to have sufficient size to hold all resulting elements.
```

  ``ends_with``, ``contains``, ``contains_subrange``, ``lexicographical_compare``.
- The sorting algorithms with device policies now can use radix sort with
  the ``less`` and ``greater`` comparators from the ``std::ranges`` namespace [#fnote1]_.
- Implemented support for bounded output for range-based set algorithms with CPU policies,
  with optional support for C++26 semantics of return values.
- Improved performance of ``sort``, ``stable_sort``, ``sort_by_key``, and ``stable_sort_by_key`` when using
  Radix sort [#fnote1]_ and device policies for key and/or value types larger than four bytes.
- Improved performance of ``inclusive_scan``, ``exclusive_scan``, ``transform_inclusive_scan``, and
  ``transform_exclusive_scan`` with device policies for non-trivially-copyable value types on GPU devices.
- Improved performance of the ``histogram_even`` and ``histogram_range`` algorithms with device policies for a small
  number of bins on GPU devices.

Known Issues and Limitations
----------------------------
Existing Issues
^^^^^^^^^^^^^^^
See oneDPL Guide for other `restrictions and known limitations`_.

> - When used with device policies. ``set_union``, ``set_intersection``, ``set_difference``, ``set_symmetric_difference``

Reviewer 2: ```suggestion
- When used with device policies, ``set_union``, ``set_intersection``, ``set_difference``, ``set_symmetric_difference``
```

  ``ends_with``, ``contains``, ``contains_subrange``, ``lexicographical_compare``.
- The sorting algorithms with device policies now can use radix sort with
  the ``less`` and ``greater`` comparators from the ``std::ranges`` namespace [#fnote1]_.
- Implemented support for bounded output for range-based set algorithms with CPU policies.

> - Implemented C++26 semantic for range-based set algorithms with CPU policies.

Reviewer: ```suggestion
- Implemented support for bounded output for range-based set algorithms with CPU policies,
  with optional support for C++26 semantics of return values.
```
